src/materials/clipping.ts

```diff
@@ -1,14 +1,65 @@
-import { Box3, Matrix4, Vector3 } from 'three';
+import { BoxGeometry, CylinderGeometry, Matrix4, Mesh, PlaneGeometry, Quaternion, SphereGeometry, Vector3 } from 'three';
 
 export enum ClipMode {
   DISABLED = 0,
   CLIP_OUTSIDE = 1,
-  HIGHLIGHT_INSIDE = 2,
+  CLIP_INSIDE = 2,
+  HIGHLIGHT_INSIDE = 3,
+  CLIP_OUTSIDE_PICK = 4,
 }
 
 export interface IClipBox {
-  box: Box3;
-  inverse: Matrix4;
   matrix: Matrix4;
-  position: Vector3;
 }
+
+export type ClipBox = {
```

Comment:
Why is this a `type` rather than an `interface`? Imho, we should be consistent and use interface everywhere. Also, most of these properties could be made `readonly`.

Reply:
Maybe we should even extract these types/interfaces into actual classes, each in their separate file.